### ISSUE TITLE
is the install.py script still relevant given data_files support in `setup.py`?

### ISSUE THREAD

**Issue by yifanwu** [jupyter/echo_kernel#12]
Hi there,

Thank you for creating this reference custom kernel!

Through some investigations, I thought that the recommended path is to use data_files support in `setup.py`, as opposed to manually moving files into the `share/jupyter/kernels/` directory. I also noticed that the `kernel.json` pattern (mentioned in https://jupyter-client.readthedocs.io/en/stable/kernels.html) is not used here.

Is this current repo not up to date with the most recent ipykernel developments, or did I mis-read some instructions?

Thanks so much for the help!

**Comment by ceteri** [comment]
I agree @yifanwu as far as I can tell this no longer installs correctly.  Am I missing something here, or are the Jupyter docs out of sync with the committed code and both out of sync with the demo repos?

I've tried with v1.1, using Py 3.10.11 on macOS:

```
IPython          : 8.22.1
ipykernel        : 6.29.3
ipywidgets       : not installed
jupyter_client   : 8.6.0
jupyter_core     : 5.7.1
jupyter_server   : 2.12.5
jupyterlab       : 4.1.2
nbclient         : 0.9.0
nbconvert        : 7.16.1
nbformat         : 5.9.2
notebook         : not installed
qtconsole        : not installed
traitlets        : 5.14.1
```

**Comment by blink1073** [comment]
The echo kernel is using [hatch](https://github.com/jupyter/echo_kernel/blob/b483ceb1b50425bf37157ddc1e0f338e21917647/pyproject.toml#L50) to install the kernel, which works on PyPI and conda-forge. install.py is still there but could be removed, simplified, and absorbed into [jupyter/echo_kernel@main/hatch_build.py](https://github.com/jupyter/echo_kernel/blob/main/hatch_build.py?rgh-link-date=2024-03-02T22%3A13%3A09Z).

**Comment by blink1073** [comment]
#17 is part of what we need, the other is https://github.com/jupyter/jupyter_client/issues/1012

**Comment by blink1073** [comment]
https://github.com/jupyter/jupyter_client/pull/1013

**Comment by ceteri** [comment]
thank you for the help @blink1073
